docs/CONTRIBUTING.md

```diff
@@ -108,36 +108,78 @@ bibtex: # citation(s) for this dataset in BibTeX format
+* `SMILES#` gets you the data of the sampled SMILES row. The `#` is used to get the corresponding data.
+* `mutagenic#no &NULL` gets you the data with `#` and recodes it. The recoding options are separated with a `&`. In this example the binary variable `mutagenic` that can be `0` or `1` gets recoded to `no ` and `NULL`. `NULL` is a "reserved word" an indicates [no value](https://en.wikipedia.org/wiki/Null_(SQL)). Thus, the `no ` gets added in front of the `mutagenic__names__adjective` if `mutagenic# == 0`.
+* `mutagenic__names__adjective` gets you from the `id` `mutagenic` the adjective names.
+
+#### Example text template 2 (mainly used for KG data)`
+`The {node1_type#} {node1_name#|node1_smiles#} {rel1_type#} the {node2_type#} {node2_protein_names#} which {rel2_type#} the {node3_type#} {node3_name#}.`
+* `node1_name#|node1_smiles#` chains together two columns with `|` so they are jointly sampled for this position. In this case we sample from the name or the SMILES representation.
+* A similar setup can be used in a single column: For `node2_protein_names` the column row can include several protein names also separated by a `|`, e.g., `Pyruvate dehydrogenase E1 component subunit beta, mitochondrial|PDHE1-B` which then samples from `Pyruvate dehydrogenase E1 component subunit beta, mitochondrial` or `PDHE1-B`.
+
+#### Example text templates 3 for multiple choice setups
+Multiple choice setups are also supported. For this we need three components:
+* `%multiple_choice_enum%2%aA1` can be used to list the multiple choice enumerations, i.e., `1, 2, or 3`, `A or B`, etc., The second `%` starts the multiple choice number sequence. Single integers and a range consisting of two integers separated by a `-` are supported to set the lower and higher number, e.g., `2-5` will sample a value between those values including the boundaries for the answer options. The third `%` is used to subselect multiple choice enumerations, i.e., `a` for lower case alphabetical enumerations, `A` for upper case alphabetical, and `1` for numerical enumerations.
+* `mutagenic%` is used to list the multiple choice enumerations with the corresponding possible answer options after the multiple choice enumerations, and
+* `%multiple_choice_result` is used to get the multiple choice enumeration of the answer, i.e., `1`, `c`.
+Please pay attention to the `%` symbol and its position as this is used to parse the different control elements from the text template.
+The sampling procedure incorporates a range of different multiple choice enumerations that are sampled randomly:
+* numerical (`1, 2, 3, ...`) and alphabetical (`a, b, c, ...` or `A, B, C, ...`) enumerations combined with
+* different suffixes, i.e., ` ` (no suffix), `.`, `.)`, `)`, and `:`, to create a range of different multiple choice enumerations.
+If only the choices `0` or `1` are available they will be recoded with `False` and `True`.
```

Comment:
What was our conclusion on experimenting if we want to enumerate all permutations of enumerations and texts? The motivation was that there might be different prior probabilities for different enumerations and we could marginalize over them. 

It is probably a very minor effect though